com.sap.cloud.lm.sl.cf.process/src/main/java/com/sap/cloud/lm/sl/cf/process/steps/DetermineDesiredStateAchievingActionsStep.java

```diff
@@ -46,24 +55,39 @@ protected StepPhase executeStep(ExecutionWrapper context) {
     }
 
     private StepPhase attemptToExecuteStep(ExecutionWrapper execution) {
-        CloudApplication app = StepsUtil.getApp(execution.getContext());
-        ApplicationStartupState currentState = computeCurrentState(execution, app);
-        getStepLogger().debug(Messages.CURRENT_STATE, app.getName(), currentState);
+        String appName = StepsUtil.getApp(execution.getContext())
+            .getName();
+        CloudControllerClient client = execution.getControllerClient();
+        CloudApplication app = client.getApplication(appName);
+        ApplicationStartupState currentState = computeCurrentState(app);
+        getStepLogger().debug(Messages.CURRENT_STATE, appName, currentState);
         ApplicationStartupState desiredState = computeDesiredState(execution.getContext(), app);
-        getStepLogger().debug(Messages.DESIRED_STATE, app.getName(), desiredState);
+        getStepLogger().debug(Messages.DESIRED_STATE, appName, desiredState);
 
         Set<ApplicationStateAction> actionsToExecute = getActionsCalculator(execution.getContext()).determineActionsToExecute(currentState,
             desiredState);
-        getStepLogger().debug(Messages.ACTIONS_TO_EXECUTE, app.getName(), actionsToExecute);
+        addStageStateIfNeeded(client, app, actionsToExecute);
+        getStepLogger().debug(Messages.ACTIONS_TO_EXECUTE, appName, actionsToExecute);
 
         StepsUtil.setAppStateActionsToExecute(execution.getContext(), actionsToExecute);
         return StepPhase.DONE;
     }
 
-    private ApplicationStartupState computeCurrentState(ExecutionWrapper execution, CloudApplication app) {
-        CloudControllerClient client = execution.getControllerClient();
+    private void addStageStateIfNeeded(CloudControllerClient client, CloudApplication cloudApplication,
+        Set<ApplicationStateAction> actionsToExecute) {
+        PlatformType platformType = configuration.getPlatformType();
+        if (platformType != PlatformType.CF) {
```

Comment:
I take it that this is not implemented for XSA, which is why you had the `getDefaultApplicationStager` method in the first place?
If that's the case, we have two options:
1. Make some minimal implementation in the XsApplicationStager that always returns `true` and track a backlog item for a proper implementation.
2. Actually implement the method in the XsApplicationStager.

I would prefer the second option if it's not too much effort. Either way, though, this `if` should not exist.